### PR TITLE
Check loaded login-as cookie when when getting web apps for a user

### PR DESCRIPTION
**Pull request by AddisonDunn**
## Product Description

In https://dimagi.atlassian.net/browse/SUPPORT-19944, users were seeing all apps upon opening Web Apps instead of the limited list defined by their role's permissions. This was because Web Apps was being loaded as their "login as" user (which had the default mobile worker role, which has access to all web apps) instead of their normal web user, even though they had turned the "limit login as" permission off for the web user's role. This PR fixes that.

## Technical Summary

Web apps loaded as a "login as" user because of a cookie named `restoreAs`. This PR rewrites the code to check if the user can still "login as" that user before utilizing the cookie, and deleting the cookie if not.

This fix also adds a sort of informal limit of being able to "log in as" 50 different users. If this limit were to be passed the cookie would be deleted--not sure the effect that would have.

## Feature Flag

Related but not entirely limited to the RESTRICT_LOGIN_AS feature flag.

### Safety story

- Tested locally and on staging: used web apps with login as, without login as, and tested the acute issue (turning off the "limited login as" permission and then immediately using web apps) and everything looks good.
- This is a relatively small change, which is why this doesn't seem worth asking for QA.

### Automated test coverage

No related automated tests exist that I'm aware of.

### QA Plan

None. Did some quick tests locally and on staging.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by MartinRiese — APPROVED** [review]
Looks good to me. But I would like at least one more set of eyes on this.

**Review by Robert-Costello — APPROVED** [review]
Looks good to me. I don't think it would hurt to have this on staging for a while, but not blocking.

**Comment by orangejenny** [comment]
The docs say that you can use space-separated values to support multiple email addresses, but [the other limited login as query](https://github.com/dimagi/commcare-hq/blob/0382aca24096ea528aaf36d4fdbf200844c83e03/corehq/apps/cloudcare/views.py#L146-L153) has a limit of 1.

File: corehq/apps/cloudcare/views.py [hunk 1]
```diff
@@ -126,31 +126,38 @@ def get_restore_as_user(request, domain):
...
-        def set_cookie(response):  # set_coookie is a noop by default
+        def set_cookie(response):  # set_cookie is a noop by default
             return response
 
+        def _get_login_as_user_query():
+            return login_as_user_query(
+                domain,
+                request.couch_user,
+                search_string='',
+                limit=500,
```
Comment (esoergel) [inline corehq/apps/cloudcare/views.py:137]: Should this go back to 1 now?
```suggestion
                limit=1,
```

**Review by esoergel — APPROVED** [review]

**Pull request merged**